.github/workflows/goreleaser.yaml

```diff
@@ -0,0 +1,101 @@
+on: [push, pull_request]
+name: goreleaser
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [1.16.x, 1.17.x]
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version }}
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade iso8601 requests securesystemslib six tuf
+    - name: Format Unix
+      if: runner.os == 'Linux'
+      run: test -z $(go fmt ./...)
+    - name: Test
+      run: go test -race -covermode atomic -coverprofile='profile.cov' ./...
+    - name: Send coverage
+      if: runner.os == 'Linux'
+      env:
+        COVERALLS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        GO111MODULE=off go get github.com/mattn/goveralls
+        $(go env GOPATH)/bin/goveralls -coverprofile=profile.cov -service=github
+    - name: Vet
+      run: go vet ./...
+    - name: Install staticcheck
+      run: "go install honnef.co/go/tools/cmd/staticcheck@v0.2.2"
+    - name: Run staticcheck
```

Comment:
we can use golangci-lint tool instead of these lines, WDYT?

Reply:
that makes sense. These lines are migrated from the old build workflow.. @asraa  @trishankatdatadog  are you fine with using golangci-lint instead of staticcheck (golangci-lint calls staticcheck, but can do more).